### PR TITLE
HC-7 / HC-6 Register / Login component

### PR DESCRIPTION
**Pull request by kj-ninja**
also added formik and yup libs

### CODE REVIEW

File: client/src/pages/auth/Login.tsx [hunk 1]
```diff
@@ -1,7 +1,67 @@
...
+      email: Yup.string().email('Please enter a valid email address.').required('Required.'),
+      password: Yup.string().required('Required.').min(8, 'At least 8 characters'),
+    }),
+    validateOnChange: true,
+    onSubmit: async (values) => {
+      // todo add errors handler
+      await firebase.auth.signInWithEmailAndPassword(values.email, values.password);
+      // todo add toast component with success message
+    },
+  });
```
Comment (JaCoWymowny) [inline client/src/pages/auth/Login.tsx:24]: isn't it a better practice to extract the validation to a separate file?

File: client/src/pages/auth/Register.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+      email: Yup.string().email('Please enter a valid email address.').required('Required.'),
+      password: Yup.string().required('Required.').min(8, 'At least 8 characters'),
+    }),
+    validateOnChange: true,
+    onSubmit: async (values) => {
+      // todo add errors handler
+      await firebase.auth.createUserWithEmailAndPassword(values.email, values.password);
+      // todo add toast component with success message
+    },
+  });
```
Comment (JaCoWymowny) [inline client/src/pages/auth/Register.tsx:24]: the same question about validate

File: client/src/pages/auth/Login.tsx (hunk 1 above)
Comment (kj-ninja) [reply client/src/pages/auth/Login.tsx:24]: mostly yes but in this case we are using login only once like register so logic could be placed in component

File: client/src/pages/auth/Register.tsx (hunk 2 above)
Comment (kj-ninja) [reply client/src/pages/auth/Register.tsx:24]: same here

**Pull request merged**